In [89]:
import matplotlib.pyplot as plt
%matplotlib inline
import pandas as pd
import numpy as np
import itertools
from sklearn.metrics import confusion_matrix
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.metrics import accuracy_score
from sklearn.linear_model import PassiveAggressiveClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.naive_bayes import BernoulliNB
from sklearn.tree import DecisionTreeClassifier

In [90]:
data=pd.read_csv('news.csv',header=0)
data.shape

(6335, 4)

In [91]:
labels=data.label
labels.head()


0    FAKE
1    FAKE
2    REAL
3    FAKE
4    REAL
Name: label, dtype: object

In [92]:
data.columns

Index(['Unnamed: 0', 'title', 'text', 'label'], dtype='object')

In [93]:
categories=list(data.label)
print(categories)

['FAKE', 'FAKE', 'REAL', 'FAKE', 'REAL', 'FAKE', 'FAKE', 'REAL', 'REAL', 'REAL', 'REAL', 'REAL', 'FAKE', 'FAKE', 'REAL', 'REAL', 'FAKE', 'FAKE', 'REAL', 'REAL', 'REAL', 'FAKE', 'REAL', 'REAL', 'FAKE', 'REAL', 'REAL', 'REAL', 'REAL', 'FAKE', 'REAL', 'FAKE', 'FAKE', 'FAKE', 'FAKE', 'FAKE', 'FAKE', 'FAKE', 'FAKE', 'FAKE', 'REAL', 'REAL', 'FAKE', 'FAKE', 'REAL', 'FAKE', 'REAL', 'REAL', 'FAKE', 'FAKE', 'REAL', 'REAL', 'FAKE', 'FAKE', 'FAKE', 'REAL', 'FAKE', 'REAL', 'FAKE', 'REAL', 'FAKE', 'REAL', 'REAL', 'FAKE', 'REAL', 'REAL', 'REAL', 'REAL', 'REAL', 'FAKE', 'REAL', 'FAKE', 'REAL', 'REAL', 'FAKE', 'REAL', 'REAL', 'REAL', 'FAKE', 'FAKE', 'REAL', 'REAL', 'REAL', 'FAKE', 'REAL', 'REAL', 'REAL', 'REAL', 'REAL', 'REAL', 'REAL', 'FAKE', 'FAKE', 'REAL', 'REAL', 'FAKE', 'FAKE', 'REAL', 'REAL', 'REAL', 'REAL', 'FAKE', 'REAL', 'REAL', 'REAL', 'REAL', 'FAKE', 'FAKE', 'FAKE', 'FAKE', 'REAL', 'REAL', 'FAKE', 'FAKE', 'FAKE', 'REAL', 'REAL', 'FAKE', 'FAKE', 'REAL', 'REAL', 'FAKE', 'REAL', 'FAKE', 'FAKE',

In [94]:
train=data['text']
test=data['label']
print(len(train))
print(len(test))

6335
6335


In [95]:
from sklearn.pipeline import make_pipeline

x_train,x_test,y_train,y_test=train_test_split(data['text'], labels, test_size=0.2, random_state=7)

In [96]:
models=make_pipeline(TfidfVectorizer(),PassiveAggressiveClassifier())

In [97]:
models.fit(x_train,y_train)


D:\anaconda\lib\site-packages\sklearn\linear_model\stochastic_gradient.py:166: FutureWarning: max_iter and tol parameters have been added in PassiveAggressiveClassifier in 0.19. If both are left unset, they default to max_iter=5 and tol=None. If tol is not None, max_iter defaults to max_iter=1000. From 0.21, default max_iter will be 1000, and default tol will be 1e-3.
  FutureWarning)


Pipeline(memory=None,
     steps=[('tfidfvectorizer', TfidfVectorizer(analyzer='word', binary=False, decode_error='strict',
        dtype=<class 'numpy.float64'>, encoding='utf-8', input='content',
        lowercase=True, max_df=1.0, max_features=None, min_df=1,
        ngram_range=(1, 1), norm='l2', preprocessor=None, smooth...=None, shuffle=True, tol=None,
              validation_fraction=0.1, verbose=0, warm_start=False))])

In [98]:

prediction=models.predict(x_test)
print(accuracy_score(y_test,prediction))

0.9344909234411997


Decision Tree accuracy_score=81.37%

NAIVE bayes BernouliiNB classfier have 
accuracy_score=80.4%

#PAC ACCURACY SCORE=93.45%

#LOGISTIC REGRESSION ACCURACY=91%

prediction=models.predict(x_test)
print(accuracy_score(y_test,prediction))

In [99]:
def prediction(s,train=x_train,models=models):
    pred=models.predict([s])
    return labels[pred]

In [100]:
prediction('Daniel Greenfield, a Shillman Journalism Fellow at the gold Center, is a New rekt writer focusing on radical  ')

FAKE    NaN
Name: label, dtype: object